formats/json/commonMain/src/kotlinx/serialization/json/JsonElement.kt

```diff
@@ -49,33 +49,57 @@ public sealed class JsonPrimitive : JsonElement() {
     public override fun toString(): String = content
 }
 
-/**
- * Creates [JsonPrimitive] from the given boolean.
- */
+/** Creates a [JsonPrimitive] from the given boolean. */
 public fun JsonPrimitive(value: Boolean?): JsonPrimitive {
     if (value == null) return JsonNull
     return JsonLiteral(value, isString = false)
 }
 
-/**
- * Creates [JsonPrimitive] from the given number.
- */
+/** Creates a [JsonPrimitive] from the given number. */
 public fun JsonPrimitive(value: Number?): JsonPrimitive {
     if (value == null) return JsonNull
     return JsonLiteral(value, isString = false)
 }
 
 /**
- * Creates [JsonPrimitive] from the given string.
+ * Creates a numeric [JsonPrimitive] from the given [UByte].
+ *
+ * The value will be encoded as a JSON number.
+ */
+@ExperimentalSerializationApi
+public fun JsonPrimitive(value: UByte): JsonPrimitive = JsonPrimitive(value.toULong())
```

Comment:
I believe we shouldn't, and nullability for other functions is rather a mistake.
It will cause unnecessary boxing, while bringing no benefit whatsoever: for nullable types, `Nothing?` overload will be chosen

Reply:
I tried adding nullability, but I ran into issues: https://github.com/Kotlin/kotlinx.serialization/issues/2130#issuecomment-1385918589

Reply:
@qwwdfsad, shouldn't we leave an explanatory comment so that in the future there will be no questions why a different approach is used?

Reply:
Would be nice to have